### ISSUE TITLE
Use pinned memory for doing transfers

### ISSUE THREAD

**Issue by coreylowman** [coreylowman/cudarc#80]

**Comment by emchristiansen** [comment]
Hey, just to be clear, you mean something like [RustaCUDA's LockedBuffer](https://github.com/bheisler/RustaCUDA/blob/c74efcb14fa24d53ac5ca1ce57e6f5794b62ce25/src/memory/locked.rs#L14), right?

FYI I need this to switch from RustaCUDA to cudarc.

**Comment by coreylowman** [comment]
Yep, this will be utilizing [cuMemAllocHost](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1gdd8311286d2c2691605362c689bc64e0) and [cuMemFreeHost](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1g62e0fdbe181dab6b1c90fa1a51c7b92c) underneath the hood, which is also what the LockedBuffer is using.

As far as how this would be integrated into result/safe APIs, we could either:
1. Have a separate htod copy method for pinned transfers (htod_pinned_copy)
2. Change the existing htod_copy methods to accept something like `AsRef<[T]>` and make sure we can implement that trait for the page locked memory.

**Comment by 0x5459** [comment]
hello.
Any progress for this issue?
Could i send a PR for this?

**Comment by tie-pilot-qxw** [comment]
Do you know if this feature has been supported? And how should I pin a certain array? I'm binding my Rust code with my Cuda code, so I really need the support to pin memory in safe Rust code.

**Comment by JordanLloydHall** [comment]
@coreylowman I love your work on this project, thank you for making Rust <-> CUDA interop possible. Do we have any status or roadmap on this feature?

**Comment by coreylowman** [comment]
Done in #336
